torchprime/utils/profiling.py

```diff
@@ -0,0 +1,19 @@
+"""Profiling utilities for trainer configuration."""
+
+from omegaconf import DictConfig
+
+
+def ensure_profile_end_step(config: DictConfig, default_span: int = 7) -> None:
```

Comment:
For the best stability in the E2E tests, I would suggest defaulting to profiling at least 10 steps. If we don't record enough steps, it might cause too much jitter and make the E2E test flaky. Also this `default_span` can be a module-level constant since it's never overridden.